cumulusci/tasks/bulkdata/load.py

```diff
@@ -191,6 +191,8 @@ def _get_columns(self, mapping):
         # If we're using Record Type mapping, `RecordTypeId` goes at the end.
         if "RecordTypeId" in columns:
             columns.remove("RecordTypeId")
+        if "RecordType" in columns:
+            columns.remove("RecordType")
```

Comment:
I didn't quite follow this:

> Never save a column named RecordType to Salesforce. It's presumably just there to represent the...actual record type. This prevents me from having to use one mapping file to generate the database tables and a different one to do the actual load.

I remember that Snowfakery supports using RecordType to refer to a record type by developer name rather than Id, but I forget how that actually gets loaded.

Reply:
If Snowfakery generates the mapping.yml then the generated mapping.yml will filter out the RecordType column, but it will use it in a WHERE clause like this:

https://github.com/SalesforceFoundation/NPSP/compare/feature/pprescod/generate_and_load_test_data?expand=1#diff-1641945226b0d2a6f8cfecdfd177bb00R34

In one of my use cases, the mapping file is being fed in as input for reasons that I probably don't need to go into in this PR. This means that mapping file has a RecordType column declared which initializes the correct database column. CCI subsequently tries to generate a field with that name.

Reply:
So it's because the same mapping file is used both by Snowfakery to initialize the db (where we need to populate RecordType) and by CCI to load it (where we don't want to include RecordType as a mapped field).

It feels like a bit of a leaky abstraction for CCI to hardcode the fact that something else will be using this specific standard-fieldname-that-is-not-actually-a-standard-field. But I'm not thinking of an alternate proposal at the moment. Can you please at least make sure there's a comment to remind us what use case this is for?

Reply:
This line is also missing coverage. You could cheat and put this removal inside a loop over the list of names to remove.

Reply:
It's a little bit deeper than just the leaky abstraction. We do have a plan to have a first-class way of specifying RecordTypes by name rather than ID.

https://salesforce.quip.com/alywAQ9UBe7G

So this is a first step towards that special treatment as opposed to just a hack relating to Snowfakery. In fact, when I finish adding it to CCI, I'll probably be able to rip record type smarts out of Snowfakery and stop using a mapping file in this case.

I did notice, by the way, that there is one (and only one) sObject that actually does use a field named RecordType, but the sObject is read-only:

https://developer.salesforce.com/docs/atlas.en-us.sfFieldRef.meta/sfFieldRef/salesforce_field_reference_CombinedAttachment.htm